### ISSUE TITLE
New Rule: static method style

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#15503]
### Rule details

forbid defining a static method using class field syntax

### Related ECMAScript feature

class fields

### What type of rule is this?

Enforces a formatting/stylistic preference

### Example code

```js
// incorrect
class Foo {
  static f = () => {}
}

// correct
class Foo {
  static f () {}
}
```

### Why should this rule be in the core instead of a plugin?

This helps keep code style consistent.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

This rule is similar to func-style https://eslint.org/docs/rules/func-style

**Comment by nzakas** [comment]
Thanks for the suggestion. We actually aren't adding any more rules that just enforce a stylistic preference. You can always create a custom rule that does this if you find it helpful.
